src/spam-protection/carding-protection-action-creator.ts

```diff
@@ -0,0 +1,71 @@
+import { createAction, ThunkAction } from '@bigcommerce/data-store';
+import { createScriptLoader } from '@bigcommerce/script-loader';
+import { concat, defer, of } from 'rxjs';
+import { catchError, switchMap, take } from 'rxjs/operators';
+
+import { InternalCheckoutSelectors } from '../checkout';
+import { throwErrorAction } from '../common/error';
+
+import { createSpamProtection } from './';
+import { SpamProtectionChallengeNotCompletedError, SpamProtectionFailedError } from './errors';
+import GoogleRecaptcha from './google-recaptcha';
+import { SpamProtectionAction, SpamProtectionActionType } from './spam-protection-actions';
+import { SpamProtectionOptions } from './spam-protection-options';
+
+export default class CardingProtectionActionCreator {
+    _googleRecaptcha: GoogleRecaptcha;
+
+    constructor(
+        private _recaptchaSitekey: string
+    ) {
+        this._googleRecaptcha = createSpamProtection(createScriptLoader());
+    }
+
+    initialize(options?: SpamProtectionOptions): ThunkAction<SpamProtectionAction, InternalCheckoutSelectors> {
+        return () => concat(
+            of(createAction(SpamProtectionActionType.InitializeRequested, undefined)),
```

Comment:
I don't think you'll need to have an action creator because unlike the checkout spam protection, this payment spam protection is not a separate action that can be triggered by the user of the SDK. Instead, it is a part of the submit order/payment action so we don't need to keep track of and surface the execution statuses / errors separately. I'd take out all the `createAction` calls in this class, and perhaps rename this class as `PaymentHumanVerificationHandler` and combine it with `getGoogleRecaptchaToken`. So eventually, you can just inject an instance of the handler into `PaymentStrategyActionCreator` and use it directly. i.e.: (`handler.handle(error)`).